core/adapters/eth_tx.go

```diff
@@ -167,18 +169,27 @@ func (e *EthTx) checkEthTxForReceipt(ethTxID int64, input models.RunInput, s *st
 		minRequiredOutgoingConfirmations = e.MinRequiredOutgoingConfirmations
 	}
 
-	hash, err := getConfirmedTxHash(ethTxID, s.DB, minRequiredOutgoingConfirmations)
-
+	receipt, err := getConfirmedReceipt(ethTxID, s.DB, minRequiredOutgoingConfirmations)
 	if err != nil {
 		logger.Error(err)
 		return models.NewRunOutputError(err)
 	}
 
-	if hash == nil {
+	if receipt == nil {
 		return models.NewRunOutputPendingOutgoingConfirmationsWithData(input.Data())
 	}
+	var r gethTypes.Receipt
+	err = json.Unmarshal(receipt.Receipt, &r)
+	if err != nil {
+		logger.Debug("unable to unmarshal tx receipt", err)
```

Comment:
should we return a `NewRunOutputError` here?

Reply:
I originally had that, but then you'd need to ensure every test either used the simulated chain or mocked a valid receipt - didn't seem worth it. I think a best effort approach is fine, falling back to prior behavior

Reply:
:+1: